components/post_view/post_body_additional_content/post_body_additional_content.jsx

```diff
@@ -70,17 +70,23 @@ export default class PostBodyAdditionalContent extends React.PureComponent {
         this.generateStaticEmbed = this.generateStaticEmbed.bind(this);
         this.isLinkToggleable = this.isLinkToggleable.bind(this);
         this.handleLinkLoaded = this.handleLinkLoaded.bind(this);
-
+        const {metadata} = props.post;
+        const embedMetadata = metadata && metadata.embeds && metadata.embeds[0];
+        const link = embedMetadata && embedMetadata.url ? embedMetadata.url : Utils.extractFirstLink(props.post.message);
         this.state = {
-            link: Utils.extractFirstLink(props.post.message),
+            link,
             linkLoadError: false,
             linkLoaded: false,
         };
     }
 
     componentDidMount() {
         // check the availability of the image rendered(if any) in the first render.
-        this.loadShortenedImageLink();
+        const {metadata} = this.props.post;
+        const embedMetadata = metadata && metadata.embeds && metadata.embeds[0];
+        if (!embedMetadata || !embedMetadata.url) {
```

Comment:
I think we might be able to just change this to `if (!this.props.post.metadata) {` and do the same thing in the componentWillReceiveProps. If we have the metadata, the server will have provided the shortened link already, so I don't think the extra checks are needed.

Reply:
@hmhealey this component is mounted for any post so we need those extra checks. 
like in the screenshot below even when if it a simple text post .
<img width="1601" alt="Screenshot 2019-03-27 at 1 03 23 AM" src="https://user-images.githubusercontent.com/4973621/55027840-3e913e80-502c-11e9-8659-999df2db8835.png">

Reply:
The component is mounted, but I would've thought that `this.loadShortenedImageLink` would only do anything if there actually is a link in the post. I didn't realize that this logic even triggers when there's no link in the post